### PR TITLE
dev-python/watchdog: Add python3_7

### PR DESCRIPTION
**Pull request by AndrewAmmerlaan**
Breaking up PR into smaller parts: https://github.com/gentoo/gentoo/pull/13988

Bug: https://bugs.gentoo.org/702458
Signed-off-by: Andrew Ammerlaan <andrewammerlaan@riseup.net>
Package-Manager: Portage-2.3.81, Repoman-2.3.20

### CODE REVIEW

File: dev-python/pathtools/pathtools-0.1.2-r3.ebuild [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+DESCRIPTION="Pattern matching and various utilities for file systems paths"
+HOMEPAGE="https://pypi.org/project/pathtools/"
+SRC_URI="mirror://pypi/p/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~x86"
+
+distutils_enable_tests nose
```
Comment (flappyports) [inline dev-python/pathtools/pathtools-0.1.2-r3.ebuild]: This does not run any tests as the source has none.

**Comment by flappyports** [comment]
@AndrewAmmerlaan I am really trying to merge some of your PR's, but the history is bloated and often I am finding things such as pathtools not having tests but you enabled them.

**Comment by AndrewAmmerlaan** [comment]
> but the history is bloated

How do I fix that? I already squashed all commits, to make the PR's more compact. I'm sorry, but I do not have a lot of experience with git, so I'm not sure what the problem is and how to fix it. Could you please explain what exactly it is that you would like me to do?

> and often I am finding things such as pathtools not having tests but you enabled them.

I'm a bit confused regarding the tests, I don't think I quite understand how you see if a package does or does not have tests. For example, I seem to have missed tests in [qtawesome](https://github.com/gentoo/gentoo/pull/14007) but I have them here where they shouldn't be.

Up to now, I've just been looking at the source code. I check if there is a 'test' or 'tests' directory present and if there is 'tests_require' or 'extras_require' in setup.py. But if I look at the [github page of qtawesome](https://github.com/spyder-ide/qtawesome) I find nothing mentioning tests in setup.py. And if I look at the [github page of pathtools](https://github.com/gorakhargosh/pathtools) I find that it has a 'tests' directory with a .py file in it. Yet, qtawesome does have tests while pathtools does not. Therefore, my logic in determining wether or not I should add `distutils_enable_tests` is clearly flawed, which leads me to do it incorrectly here and in my other PR's. Could you please explain to me how it is you see wether a package has tests or not?

I'm currently away for the Christmas holidays, but I'll try to fix everything when I get back next week.

File: dev-python/pathtools/pathtools-0.1.2-r3.ebuild (hunk 1 above)
Comment (AndrewAmmerlaan) [reply dev-python/pathtools/pathtools-0.1.2-r3.ebuild]: Fixed

**Comment by AndrewAmmerlaan** [comment]
>New issues caused by PR:
> https://qa-reports.gentoo.org/output/gentoo-ci/7d86594/output.html#net-misc/crossbar

The bot is complaining about net-misc/crossbar-18.3.1 because it requires`>=dev-python/watchdog-0.8.3[${PYTHON_USEDEP}]` with py3_5, however as far as I can see that dependency is still satisfied by watchdog-0.8.3. Should I just let i be?

**Comment by flappyports** [comment]
> > New issues caused by PR:
> > https://qa-reports.gentoo.org/output/gentoo-ci/7d86594/output.html#net-misc/crossbar
> 
> The bot is complaining about net-misc/crossbar-18.3.1 because it requires`>=dev-python/watchdog-0.8.3[${PYTHON_USEDEP}]` with py3_5, however as far as I can see that dependency is still satisfied by watchdog-0.8.3 (I can still `emerge -1 crossbar` without problems).
> 
> Should I just let i be?
> I could also re-add python2_7 and python3_5 to watchdog and pathtools to make the bot happy?

Yes, please add py2.7 and py3.5 back to make CI happy. otherwise the tree is broken. We will have to remove it later when we do a mass removal of py2 and py35

**Comment by AndrewAmmerlaan** [comment]
> Yes, please add py2.7 and py3.5 back to make CI happy. otherwise the tree is broken. We will have to remove it later when we do a mass removal of py2 and py35

Alright, fixed

File: dev-python/watchdog/watchdog-0.9.0.ebuild [hunk 2]
```diff
@@ -15,27 +15,21 @@ LICENSE="Apache-2.0"
...
 	dev-python/argh[${PYTHON_USEDEP}]
 	dev-python/pathtools[${PYTHON_USEDEP}]"
 DEPEND="${CDEPEND}
-	test? (
-		dev-python/pytest[${PYTHON_USEDEP}]
-		>=dev-python/pytest-timeout-0.3[${PYTHON_USEDEP}]
-	)"
+	test? ( >=dev-python/pytest-timeout-0.3[${PYTHON_USEDEP}] )"
+
+distutils_enable_tests pytest
```
Comment (flappyports) [inline dev-python/watchdog/watchdog-0.9.0.ebuild]: Great job recognizing the eclass replacements in this ebuild!

File: dev-python/pathtools/pathtools-0.1.2-r3.ebuild [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+inherit distutils-r1
+
+DESCRIPTION="Pattern matching and various utilities for file systems paths"
+HOMEPAGE="https://pypi.org/project/pathtools/"
+SRC_URI="mirror://pypi/p/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~x86"
```
Comment (flappyports) [inline dev-python/pathtools/pathtools-0.1.2-r3.ebuild]: This package does have docs in the ebuild which should be built, but it is not a regression. Will merge. Please check next time to correct the previous devs lack of enabling docs.

**Pull request closed without merging**